Sources/Apollo/NetworkTransport.swift

```diff
@@ -9,5 +9,22 @@ public protocol NetworkTransport {
   ///   - error: An error that indicates why a request failed, or `nil` if the request was succesful.
   /// - Returns: An object that can be used to cancel an in progress request.
   func send<Operation>(operation: Operation, completionHandler: @escaping (_ response: GraphQLResponse<Operation>?, _ error: Error?) -> Void) -> Cancellable
+
+  /// Send a GraphQL operation to a server and return a response.
+  ///
+  /// - Parameters:
+  ///   - operation: The operation to send.
+  ///   - files: A list of files to send as a multipart request.
+  ///   - progressHandler: A closure to call periodically as the request is sent.
+  ///   - completionHandler: A closure to call when a request completes.
+  ///   - response: The response received from the server, or `nil` if an error occurred.
+  ///   - error: An error that indicates why a request failed, or `nil` if the request was succesful.
+  /// - Returns: An object that can be used to cancel an in progress request.
+  func upload<Operation: GraphQLOperation>(operation: Operation, files: [GraphQLFile]?, progressHandler: ((Progress) -> Void)?, completionHandler: @escaping (GraphQLResponse<Operation>?, Error?) -> Void) -> Cancellable
```

Comment:
This is probably the biggest thing: Does this actually need to be on `NetworkTransport`, or should it be specific to `HTTPNetworkTransport`? I'd be quite surprised if anyone's sending files over websockets.

Reply:
Hmm. I don't think that. 
Maybe it would be better to only add it in `HTTPNetworkTransport` and if people wan't it in web sockets for some reason, we can always add it. 🥳